### ISSUE TITLE
menu redial via d has stopped working

### ISSUE THREAD

**Issue by juha-h** [baresip/baresip#1239]
I build baresip from Feb 23 master and noticed that menu redial by pressing `d` and return has stopped working.

**Comment by alfredh** [comment]
is there anything in the dial buffer?

**Comment by juha-h** [comment]
Alfred E. Heggestad writes:

> is there anything in the dial buffer?

Here is an example.  After `All ...` line I press `d`, cursor moves to
the right and I type `foo` followed by return.  baresip tries to place
the call as normal.

After 480 Temporalily Unavailable response I type `d` and cursor again
moves to the right.  Then I press return and nothing happens.

-----------------------------------------------------------------------

test@test.tutpro.com: (prio 0) {1/TLS/v4} 200 OK (OpenSIPg SIP Proxy (5.5.0-0b26 (x86_64/linux))) [1 binding]
All 1 useragent registered successfully! (151 ms)
>                              foo
test@test.tutpro.com: selected for request
ua: using best effort AF: af=AF_INET
call: connecting to 'sip:foo@test.tutpro.com'..
call: SIP Progress: 100 Trying (/)
call: SIP Progress: 100 Trying (/)
sip:foo@test.tutpro.com: session closed: 480 Temporalily Unavailable
>

**Comment by alfredh** [comment]
https://github.com/baresip/baresip/blob/master/modules/menu/static_menu.c

check dial-handler

is menu->dialbuf set?

**Comment by alfredh** [comment]
it looks like `&word[0]` is not set when just dialing `d`

perhaps this patch may help:

```diff
diff --git a/modules/menu/static_menu.c b/modules/menu/static_menu.c
index 3d41bf6..4266d2d 100644
--- a/modules/menu/static_menu.c
+++ b/modules/menu/static_menu.c
@@ -447,9 +447,11 @@ static int dial_handler(struct re_printf *pf, void *arg)
 
        (void)pf;
 
-       err = pl_strdup(&uri, &word[0]);
-       if (err)
-               return err;
+       if (pl_isset(&word[0])) {
+               err = pl_strdup(&uri, &word[0]);
+               if (err)
+                       return err;
+       }
 
        if (str_isset(uri)) {
 
```

**Comment by juha-h** [comment]
Thanks, the patch made redial to work again.  Should I do the pull request?

**Comment by alfredh** [comment]
fixed in defa7ab3b9f95383dac5100056f1618dce343819

thanks for reporting!
